core/src/main/java/com/netflix/conductor/core/reconciliation/WorkflowRepairService.java

```diff
@@ -54,14 +53,16 @@
     private final Predicate<Task> isTaskRepairable = task -> {
         if (systemTaskRegistry.isSystemTask(task.getTaskType())) {    // If system task
             WorkflowSystemTask workflowSystemTask = systemTaskRegistry.get(task.getTaskType());
-            return workflowSystemTask.isAsync() && !workflowSystemTask.isAsyncComplete(task) &&
+            return workflowSystemTask.isAsync() && (!workflowSystemTask.isAsyncComplete(task)
```

Comment:
A comment explaining what criteria of system tasks are selected for repair would help.